app/controllers/api/v1/apps/credentials_controller.rb

```diff
@@ -1,9 +1,11 @@
 # frozen_string_literal: true
 
 class Api::V1::Apps::CredentialsController < Api::BaseController
-  before_action -> { doorkeeper_authorize! :read }
-
   def show
-    render json: doorkeeper_token.application, serializer: REST::ApplicationSerializer, fields: %i(name website vapid_key)
+    if valid_doorkeeper_token?
+      render json: doorkeeper_token.application, serializer: REST::ApplicationSerializer, fields: %i(name website vapid_key client_id scopes)
```

Comment:
Is it safe to return the VAPID key if the token purposefully has a smaller scope than the application?

Reply:
This is actually kinda misleading, as it's not the VAPID Key for the user/application, but in fact it's the public key for the server. I found this kinda confusing in the response. It feels like we should just host the VAPID public key at a well known location?

But also, we don't currently support the differentiate between token permissions and application permissions.

Reply:
ohhh you're right, sorry for that!